standard/patterns.md

```diff
@@ -48,32 +59,41 @@ Each pattern form defines the set of types for input values that the pattern may
 
 Each pattern form defines the set of values for which the pattern *matches* the value at runtime.
 
+The order of evaluation of operations and side effects during pattern-matching (calls to `Deconstruct`, property accesses, and invocations of methods in `System.ITuple`) is not specified.
+
 ### 11.2.2 Declaration pattern
 
-A *declaration_pattern* is used to test that a value has a given type and, if the test succeeds, provide the value in a variable of that type.
+A *declaration_pattern* is used to test that a value has a given type and, if the test succeeds, to optionally provide the value in a variable of that type.
 
 ```ANTLR
 declaration_pattern
     : type simple_designation
     ;
 simple_designation
-    : single_variable_designation
+    : discard_designation
+    | single_variable_designation
+    ;
+discard_designation
+    : '_'
     ;
 single_variable_designation
     : identifier
     ;
 ```
 
-The runtime type of the value is tested against the *type* in the pattern using the same rules specified in the is-type operator ([§12.13.12.1](expressions.md#1213121-the-is-type-operator)). If the test succeeds, the pattern *matches* that value. It is a compile-time error if the *type* is a nullable value type ([§8.3.12](types.md#8312-nullable-value-types)). This pattern form never matches a `null` value.
+A *simple_designation* with the token `_` shall be considered a *discard_designation* rather than a *single_variable_designation*.
```

Comment:
Following the comment on [line 4419 of expressions.md](https://github.com/dotnet/csharpstandard/pull/873/files#r2361681490):

```suggestion
When recognising a *simple_designation* if both the *discard_designation* and *single_variable_designation* alternatives are applicable then the former shall be chosen.
```

Reply:
Is the suggestion really more clear?